File: gui_api/app/api/routes/event_comment.py
import json

from fastapi import APIRouter, Request, Response, status
from uuid import UUID

from api import db_api
from api.routes import helpers
from api_models.event_comment import EventCommentCreate, EventCommentRead, EventCommentUpdate


router = APIRouter(
    prefix="/event/comment",
    tags=["Event Comment"],
)


#
# CREATE
#


def create_event_comments(
    event_comments: list[EventCommentCreate],
    request: Request,
    response: Response,
):
    db_api.post(
        path="/event/comment/",
        payload=[json.loads(c.json(exclude_unset=True)) for c in event_comments],
        expected_status=status.HTTP_201_CREATED,
    )

    response.headers["Content-Location"] = request.url_for("get_event_comment", uuid=event_comments[-1].uuid)


helpers.api_route_create(router, create_event_comments)


#
# READ
#


def get_event_comment(uuid: UUID):
    return db_api.get(path=f"/event/comment/{uuid}", expected_status=status.HTTP_200_OK)


helpers.api_route_read(router, get_event_comment, EventCommentRead)


#
# UPDATE
#


def update_comment(
    uuid: UUID,
    event_comment: EventCommentUpdate,
    request: Request,
    response: Response,
):
    db_api.patch(
        path=f"/event/comment/{uuid}",
        payload=json.loads(event_comment.json()),
    )

    response.headers["Content-Location"] = request.url_for("get_event_comment", uuid=uuid)


helpers.api_route_update(router, update_comment)

File: frontend/src/services/api/eventComment.ts
import { commentCreate, commentRead, commentUpdate } from "@/models/comment";
import { UUID } from "@/models/base";
import { BaseApi } from "./base";

const api = new BaseApi();
const endpoint = "/event/comment/";

export const EventComment = {
  create: (data: commentCreate[], getAfterCreate = false): Promise<void> =>
    api.create(endpoint, data, getAfterCreate),

  read: (uuid: UUID): Promise<commentRead> => api.read(`${endpoint}${uuid}`),

  update: (uuid: UUID, data: commentUpdate): Promise<void> =>
    api.update(`${endpoint}${uuid}`, data),
};
